analyses/FL_cd_2020/03_sim_FL_cd_2020.R

```diff
@@ -0,0 +1,173 @@
+###############################################################################
+# Simulate plans for `FL_cd_2020`
+# © ALARM Project, March 2022
+###############################################################################
+
+set.seed(2020)
+
+cluster_pop_tol <- 0.0175
+nsims <- 35000
+
+# Unique ID for each row, will use later to reconnect pieces
+map$row_id <- 1:nrow(map)
+
+# Run the simulation -----
+cli_process_start("Running simulations for {.pkg FL_cd_2020}")
+
+########################################################################
+
+# Cluster #1: Southern Florida
+
+map_south <- map %>% filter(region == "South")
+map_south <- set_pop_tol(map_south, cluster_pop_tol)
+attr(map_south, "pop_bounds") <-  attr(map, "pop_bounds")
+
+########################################################################
+
+# Setup for cluster constraint
+map <- map %>%
+    mutate(cluster_edge = ifelse(row_id %in% map_south$row_id, 1, 0))
+
+z <- geomander::seam_geom(map$adj, map, admin = "cluster_edge", seam = c(0, 1))
+
+z <- z[z$cluster_edge == 1, ]
+
+border_idxs <- which(map_south$row_id %in% z$row_id)
+
+########################################################################
+
+constraints <- redist_constr(map_south) %>%
+    # reward districts with black vap % below 10% and above 40%, enforcing barrier in between
+    add_constr_grp_hinge(10, vap_black, vap, 0.4) %>%
+    add_constr_grp_hinge(-12, vap_black, vap, 0.1) %>%
+    # reward districts with hispanic vap % below 30% and above 70%, enforcing barrier in between
+    add_constr_grp_hinge(5, vap_hisp, vap, 0.7) %>%
```

Comment:
can we use `inv_hinge` here? If the goal is to keep it below, this is nudging it too high, which may contribute to the diversity issues here.

Reply:
> `inv_hinge` throughout or on any of the constraints specifically?